security/fma-withdrawals-root-in-block-header.md

```diff
@@ -0,0 +1,133 @@
+
+- **Risk Assessment:**
+
+  **High impact, low likelihood**
+
+  **Mitigations:**
+  This should be caught by cross-client testing (kurtosis and devnets)
+
+- **Detection:**
+  If the bug made it to production, replica healthcheck alerts would fire as nodes diverged.
+
+- **Recovery Path(s)**:
+  The buggy client(s) would need to be patched to bring them back into compliance with the specification.
+
+### FM3: Failure of p2p network due to bug in new topic/message serde logic
+
+- **Description:**
+  Because this feature introduces a new p2p gossip topic and message serialization format, a bug can mean the failure of p2p gossip for any chain with Isthmus active. This would cause an unsafe chain halt.
+
+- **Risk Assessment:**
+
+  **High impact, low likelihood**
+
+  **Mitigations:**
+  We rely on end-to-end testing (including fuzzing) to catch any bugs in this code path. We could run extended fuzzing campaigns.
+
+- **Detection:**
+  Continuous integration, or Kurtosis and/or devnet testing should catch this. Failing that, the bug makes it to production, our alerting infrastructure would notify us.
+
+- **Recovery Path(s)**:
+  The bug would need to be patched and new op-node release cut and rolled out.
+
+## Generic failure modes:
+
+See the [generic FMA](./fma-generic-hardfork.md):
+
+- Chain halt at activation (there is a change to the engine API, which elevates this risk)
+- Activation failure
+- Invalid setImplementation execution
+- Chain split (across clients)
+
+## Specific Action Items
+
+- [x] (BLOCKING) e2e tests must check for consistency between output roots returned from op-node and those constructed manually in the old way https://github.com/ethereum-optimism/optimism/blob/6a436fe9ac9acb215b0f4b9f87ccd3832f4d6b72/op-e2e/actions/upgrades/isthmus_fork_test.go#L286-L301
+- [ ] (non-BLOCKING) op-node could be furnished with an override to make it serve output roots in the legacy fashion; this would also aid in testing (see above item). It would even allow us to run the two systems side by side for a time before fully switching over.
+- [ ] (non-BLOCKING) op-geth could be made to log a critical error triggering an alert if ever the `withdrawals` list in the block body is non empty (post Isthmus)
+
+## Generic Action Items
+
+- [x] (BLOCKING): We have implemented extensive unit and end-to-end testing of the activation flow: https://github.com/ethereum-optimism/optimism/blob/develop/op-e2e/actions/upgrades/isthmus_fork_test.go
+- [ ] (BLOCKING): We have implemented multi-client testing to reduce the chance of bugs (the above test could be migrated to a fault proof test where it can run on kona)
+- [ ] (non-BLOCKING): We have implemented fuzz testing in a kurtosis multi-client devnet to reduce the chance of bugs
+- [ ] (BLOCKING): We will be testing the activation on our devnets and testnets.
+
```

Comment:
```suggestion
- [ ] (NON-BLOCKING): Making sure, that component that now fetch the withdrawalHash by this have also a fallback mechanism to previous Merkle Tree method in case there is a bug.
- [ ] (NON-BLOCKING): Creating a monitoring that differential testing from the merkle tree inclusion computation and the block.header request (assignee: @Ethnical)
```